// jest-node-http/lib/ServerResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ServerResponse {
    constructor() {
        this.addTrailers = jest.fn();
        this.connection = jest.fn();
        this.end = jest.fn();
        this.finished = jest.fn();
        this.getHeader = jest.fn();
        this.getHeaderNames = jest.fn();
        this.getHeaders = jest.fn();
        this.hasHeader = jest.fn();
        this.headersSent = jest.fn();
        this.removeHeader = jest.fn();
        this.sendDate = jest.fn();
        return this;
    }
    resetMocked() {
        this.addTrailers.mockReset();
        this.connection.mockReset();
        this.end.mockReset();
        this.finished.mockReset();
        this.getHeader.mockReset();
        this.getHeaderNames.mockReset();
        this.getHeaders.mockReset();
        this.hasHeader.mockReset();
        this.headersSent.mockReset();
        this.removeHeader.mockReset();
        this.sendDate.mockReset();
    }
}
exports.ServerResponse = ServerResponse;
