// jest-node-http/lib/ClientRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ClientRequest {
    constructor(options, cb) {
        this.abort = jest.fn();
        this.aborted = jest.fn();
        this.connection = jest.fn();
        this.end = jest.fn();
        this.flushHeaders = jest.fn();
        this.getHeader = jest.fn();
        this.removeHeader = jest.fn();
        this.setHeader = jest.fn();
        this.setNoDelay = jest.fn();
        this.setSocketKeepAlive = jest.fn();
        this.setTimeout = jest.fn();
        this.socket = jest.fn();
        this.write = jest.fn();
        return this;
    }
    resetMocked() {
        this.abort.mockReset();
        this.aborted.mockReset();
        this.connection.mockReset();
        this.end.mockReset();
        this.flushHeaders.mockReset();
        this.getHeader.mockReset();
        this.removeHeader.mockReset();
        this.setHeader.mockReset();
        this.setNoDelay.mockReset();
        this.setSocketKeepAlive.mockReset();
        this.setTimeout.mockReset();
        this.socket.mockReset();
        this.write.mockReset();
    }
}
exports.ClientRequest = ClientRequest;
