// jest-node-http/lib/IncomingMessage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class IncomingMessage {
    constructor() {
        this.destroy = jest.fn();
        this.headers = jest.fn();
        this.httpVersion = jest.fn();
        this.method = jest.fn();
        this.rawHeaders = jest.fn();
        this.rawTrailers = jest.fn();
        this.setTimeout = jest.fn();
        this.socket = jest.fn();
        this.statusCode = jest.fn();
        this.statusMessage = jest.fn();
        this.trailers = jest.fn();
        this.url = jest.fn();
        return this;
    }
    resetMocked() {
        this.destroy.mockReset();
        this.headers.mockReset();
        this.httpVersion.mockReset();
        this.method.mockReset();
        this.rawHeaders.mockReset();
        this.rawTrailers.mockReset();
        this.setTimeout.mockReset();
        this.socket.mockReset();
        this.statusCode.mockReset();
        this.statusMessage.mockReset();
        this.trailers.mockReset();
        this.url.mockReset();
    }
}
exports.IncomingMessage = IncomingMessage;
