// jest-node-http/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Agent_1 = require("./Agent");
const ClientRequest_1 = require("./ClientRequest");
const IncomingMessage_1 = require("./IncomingMessage");
const Methods_1 = require("./Methods");
const Server_1 = require("./Server");
const ServerResponse_1 = require("./ServerResponse");
const StatusCodes_1 = require("./StatusCodes");
function createServer(opts, requestListener) {
    return new Server_1.Server(opts, requestListener);
}
function request(options, cb) {
    return new ClientRequest_1.ClientRequest(options, cb);
}
function get(options, cb) {
    const req = request(options, cb);
    req.end();
    return req;
}
module.exports = {
    Agent: Agent_1.Agent,
    ClientRequest: ClientRequest_1.ClientRequest,
    IncomingMessage: IncomingMessage_1.IncomingMessage,
    METHODS: Methods_1.METHODS,
    STATUS_CODES: StatusCodes_1.STATUS_CODES,
    Server: Server_1.Server,
    ServerResponse: ServerResponse_1.ServerResponse,
    createServer,
    get,
    globalAgent: Agent_1.globalAgent,
    request,
};
